e2e/vault-provider.bats

```diff
@@ -0,0 +1,148 @@
+vault_container_name="vault"
+
+setup() {
+  bats_load_library bats-support
+  bats_load_library bats-assert
+
+  start_vault
+
+  setup_pod
+
+  run go build
+  assert_success
+}
+
+start_vault() {
+  docker-compose up -d
```

Comment:
the default `docker-compose` is deprecated (https://docs.docker.com/compose/install/), using docker CLI compose works
```suggestion
  docker compose up -d
```